## Cleaning Chicago Marathon Data

Data from this will be used in Tableau

In [2]:
#Pandas library used for cleaning
import pandas as pd

In [108]:
#Importing csv with pace of all runners
df = pd.read_csv('pace_testset.csv')

In [109]:
df.head()

Unnamed: 0                     Name  Place Overall sex  BIB  5k min/mile1  \
0           0  Cherono, Lawrence (KEN)              1   M    4          4.45   
1           1     Debela, Dejene (ETH)              2   M   38          4.45   
2           2     Mengstu, Asefa (ETH)              3   M    5          4.45   
3           3      Karoki, Bedan (KEN)              4   M    9          4.45   
4           4       Abdi, Bashir (BEL)              5   M   10          4.46   

   10k min/mile1  15k min/mile1  20k min/mile1  Half min/mile1  25k min/mile1  \
0           4.45           4.45           4.47            4.45           4.50   
1           4.45           4.45           4.47            4.44           4.49   
2           4.45           4.45           4.47            4.44           4.49   
3           4.44           4.45           4.47            4.44           4.49   
4           4.44           4.48           4.53            4.57           4.57   

   30k min/mile1  35k min/mile1  40k min/mile1  Finish min/mile1  \
0           4.51           4.48           4.55              4.50   
1           4.51           4.48           4.55              4.52   
2           4.51           4.48           4.55              4.53   
3           4.52           4.48           4.55              4.57   
4           5.00           4.48           4.43              4.40   

   Total Finish time  
0           2.095833  
1           2.096111  
2           2.096667  
3           2.098056  
4           2.103889

In [110]:
#Dropping unecessary column
df = df.drop(columns =['Unnamed: 0'])

In [114]:
#Importing full unencoded chicago marathon results
df2 = pd.read_csv('full_uneconded_results.csv')

/Users/johnvictorifermo/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (5,7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [115]:
df2.head()

Unnamed: 0  Unnamed: 0.1                     Name  BIB Age Group Age  \
0           0             0  Cherono, Lawrence (KEN)    4     30-34  31   
1           1             1     Debela, Dejene (ETH)   38     20-24  24   
2           2             2     Mengstu, Asefa (ETH)    5     30-34  31   
3           3             3      Karoki, Bedan (KEN)    9     25-29  29   
4           4             4       Abdi, Bashir (BEL)   10     30-34  30   

   Place gender Place Age  Place Overall Start Time  ... 40k Time of Day  \
0             1         1              1   07:30:03  ...      09:29:12AM   
1             2         1              2   07:30:03  ...      09:29:10AM   
2             3         2              3   07:30:03  ...      09:29:10AM   
3             4         1              4   07:30:03  ...      09:29:11AM   
4             5         3              5   07:30:03  ...      09:29:55AM   

   40k Time 40k Difference 40k min/mile  Finish Finish Time of Day  \
0  01:59:10          15:17        04:55  Finish         09:35:47AM   
1  01:59:08          15:15        04:55  Finish         09:35:48AM   
2  01:59:08          15:15        04:55  Finish         09:35:50AM   
3  01:59:09          15:16        04:55  Finish         09:35:55AM   
4  01:59:53          14:39        04:43  Finish         09:36:16AM   

  Finish Time.1 Finish Difference Finish min/mile sex  
0      02:05:45             06:36           04:50   M  
1      02:05:46             06:39           04:52   M  
2      02:05:48             06:40           04:53   M  
3      02:05:53             06:45           04:57   M  
4      02:06:14             06:21           04:40   M  

[5 rows x 64 columns]

In [116]:
df2 = df2[['Start Time', 'Finish Time of Day', 'Finish Time.1']]

In [117]:
df2.head()

Start Time Finish Time of Day Finish Time.1
0   07:30:03         09:35:47AM      02:05:45
1   07:30:03         09:35:48AM      02:05:46
2   07:30:03         09:35:50AM      02:05:48
3   07:30:03         09:35:55AM      02:05:53
4   07:30:03         09:36:16AM      02:06:14

In [118]:
result = pd.concat([df, df2], axis=1)

In [119]:
result.head()

Name  Place Overall sex  BIB  5k min/mile1  \
0  Cherono, Lawrence (KEN)              1   M    4          4.45   
1     Debela, Dejene (ETH)              2   M   38          4.45   
2     Mengstu, Asefa (ETH)              3   M    5          4.45   
3      Karoki, Bedan (KEN)              4   M    9          4.45   
4       Abdi, Bashir (BEL)              5   M   10          4.46   

   10k min/mile1  15k min/mile1  20k min/mile1  Half min/mile1  25k min/mile1  \
0           4.45           4.45           4.47            4.45           4.50   
1           4.45           4.45           4.47            4.44           4.49   
2           4.45           4.45           4.47            4.44           4.49   
3           4.44           4.45           4.47            4.44           4.49   
4           4.44           4.48           4.53            4.57           4.57   

   30k min/mile1  35k min/mile1  40k min/mile1  Finish min/mile1  \
0           4.51           4.48           4.55              4.50   
1           4.51           4.48           4.55              4.52   
2           4.51           4.48           4.55              4.53   
3           4.52           4.48           4.55              4.57   
4           5.00           4.48           4.43              4.40   

   Total Finish time Start Time Finish Time of Day Finish Time.1  
0           2.095833   07:30:03         09:35:47AM      02:05:45  
1           2.096111   07:30:03         09:35:48AM      02:05:46  
2           2.096667   07:30:03         09:35:50AM      02:05:48  
3           2.098056   07:30:03         09:35:55AM      02:05:53  
4           2.103889   07:30:03         09:36:16AM      02:06:14

In [120]:
#Sorting values by Overall Place
df = result.sort_values(by=['Place Overall'])

In [121]:
#Testing if it works
df.tail(50)

Name  Place Overall sex    BIB  5k min/mile1  \
44640             Garcia, Celia (MEX)          45443   F  30073         13.45   
44641             Toledo, Isela (USA)          45444   F  35171         13.45   
24084            Wilson, Andrew (USA)          45451   M  56124         27.10   
44642       Rodriguez, Angelica (USA)          45454   F  35359         11.15   
44643             Dodson, Tiara (USA)          45455   F   1770         11.41   
44644        Kondrat, Elizabeth (USA)          45457   F  37037         13.55   
44645             Kirk, Beverly (USA)          45458   F  41556         13.41   
24085         Suarez, Francisco (USA)          45463   M  20594         13.48   
24086       Kilchenmann, Stefan (SUI)          45471   M  59698         35.24   
44646      Temelkovski, Shannon (USA)          45472   F  60820         12.40   
44647            Andrade, Sonya (USA)          45476   F  34862         13.38   
24087           Randall, Joshua (USA)          45478   M  63121         14.29   
24088           Alba, Christian (USA)          45481   M    867         14.31   
24089            Emrick, Andrew (USA)          45482   M    848         14.31   
44648          Starke, Danielle (USA)          45483   F    890         14.31   
44649            Acuna, Daniela (CRC)          45490   F  24813         13.15   
24090           Nieves, Esteban (USA)          45496   M  11697         12.15   
24091        Chaikouang, Robert (USA)          45500   M  28739         11.53   
44650          Hampe, Elizabeth (USA)          45504   F  39445         13.21   
24092           Martinez, Jesus (USA)          45509   M  32576         12.23   
44651             Johnson, Beth (USA)          45512   F  33544         13.33   
44652           Mendres, Mariav (USA)          45531   F  36026         15.24   
24093                Price, Ron (USA)          45533   M  40515         12.40   
44653          Smith, Christine (USA)          45534   F  40537         12.41   
24094              Murphy, Paul (USA)          45537   M  37058         11.43   
24095          Getueza, Eduardo (USA)          45540   M  40531         12.05   
24096           Kodela, Krishna (IND)          45547   M  16908         13.01   
44654           Mullery, Alisha (USA)          45552   F  15924         15.03   
24097               Banda, Abel (USA)          45564   M  32253         11.24   
44655            Choi, Soonyeon (KOR)          45566   F  39920         11.04   
44656              Scott, Donna (USA)          45572   F  43999         15.08   
44657            Puntillo, Kimi (USA)          45576   F   7725         14.22   
24098         Samson, Christian (USA)          45582   M  28377         12.13   
44658      Falvo, Franca Oreana (ITA)          45584   F  39642         16.05   
24099          Jakubik, Richard (USA)          45600   M   5656         12.00   
44659                Haro, Alma (USA)          45604   F   7129         11.15   
44660  El Sherif Esposito, Rola (USA)          45613   F  31351         12.33   
44661         Mcintyre, Kathryn (USA)          45617   F  40663         26.42   
44662                   Sun, Yu (CHN)          45633   F  14802         12.14   
44663          Bogard, Patricia (USA)          45647   F  32380         13.59   
44664             Akins, Larken (USA)          45656   F  19519         14.33   
24100             Rivera, Aaron (MEX)          45674   M  15848         12.39   
44665           Maciel, Susanna (USA)          45675   F  57141         33.59   
44666           Munsayac, Julie (USA)          45681   F  13519         15.23   
24101        Spijkerman, Maikel (NED)          45694   M   6786         12.48   
44667      Cook-Glanton, Regina (USA)          45717   F  17383         14.07   
44668             Kalenga, Jane (ZAM)          45730   F    908         15.32   
44669         Driscoll, Natasha (USA)          45748   F    858         15.32   
44670            Byrnes, Lauren (USA)          45749   F    855 

In [122]:
#Testing if it is sorted by Place Overall
df.head(50)

Name  Place Overall sex    BIB  5k min/mile1  \
0       Cherono, Lawrence (KEN)              1   M      4          4.45   
1          Debela, Dejene (ETH)              2   M     38          4.45   
2          Mengstu, Asefa (ETH)              3   M      5          4.45   
3           Karoki, Bedan (KEN)              4   M      9          4.45   
4            Abdi, Bashir (BEL)              5   M     10          4.46   
5             Tura, Seifu (ETH)              6   M     39          4.45   
6         Chumba, Dickson (KEN)              7   M      6          4.45   
7               Farah, Mo (GBR)              8   M      1          4.46   
8            Riley, Jacob (USA)              9   M     16          5.00   
9           Mock, Jerrell (USA)             10   M     36          5.01   
10        Stinson, Parker (USA)             11   M     23          5.00   
11     Bumbalough, Andrew (USA)             12   M     20          5.00   
12          Oishi, Minato (JPN)             13   M     11          4.58   
13      Mcdonald, Matthew (USA)             14   M     21          5.01   
14           Smith, Scott (USA)             15   M     15          5.01   
15         Gregg, Brendan (USA)             16   M     17          5.01   
16           Droddy, Noah (USA)             17   M     25          5.00   
17       Given, Wilkerson (USA)             18   M     24          5.01   
18         Estrada, Diego (USA)             19   M     19          5.01   
19      Hayakawa, Tsubasa (JPN)             20   M     14          4.58   
20        Takeuchi, Ryoma (JPN)             21   M     12          5.00   
21         Shrader, Brian (USA)             22   M     18          5.01   
24102      Kosgei, Brigid (KEN)             23   F    101          4.59   
22           Monroe, Alex (USA)             24   M     34          5.01   
23        Kremske, Daniel (USA)             25   M     31          5.05   
24          Medina, Oscar (MEX)             42   M    568          5.12   
25      Campbell, Patrick (USA)             43   M    407          5.07   
26      Lustgarten, Brett (USA)             44   M     32          5.12   
27            Smith, Ryan (USA)             45   M    651          5.12   
28             Gade, Noah (USA)             46   M    465          5.12   
29             Bang, Eric (CAN)             47   M    370          5.12   
30         Eaton, Michael (USA)             48   M     29          5.21   
31          Thomas, Jason (USA)             49   M    665          5.21   
32          Nally, Daniel (USA)             50   M    580          5.21   
24103    Yeshaneh, Ababel (ETH)             51   F    106          5.02   
24104       Burka, Gelete (ETH)             52   F    102          5.10   
33      Craighead, Daniel (USA)             53   M    430          5.19   
34       Hillard, Matthew (USA)             54   M    497          5.21   
35          Chorney, Matt (USA)             55   M    417          5.22   
36      Bramblett, Jordan (USA)             57   M  13197          5.04   
37        Erskine, Andrew (USA)             58   M    457          5.21   
38          Nelson, Aaron (USA)             59   M    582          5.15   
39        Burkhardt, Glen (USA)             60   M    402          5.00   
40        Guillen, Ramiro (USA)             61   M    475          5.08   
41        Ripley, Zachary (USA)             62   M    619          5.23   
42        Barker, Zachary (USA)             63   M    371          5.15   
43         Hall, Benjamin (USA)             64   M    481          5.22   
44     Rylaarsdam, Andrew (USA)             65   M    713          5.22   
45          Rayner, Kenny (USA)             66   M    613          5.24   
46           Ruth, Marius (NOR)             67   M   5060          5.19   

       10k min/mile1  15k min/mile1  20k min/mile1  Half min/mile1  \
0               4.45           4.45           4.47            4.45   
1               4.45           4.45           4.47           

In [136]:
#Saving into a final csv
df.to_csv("final.csv")

In [123]:
df.shape

(44672, 18)

There are 18 features and over 44,672 runners in the Chicago 2019 Marathon

**Some exploratory analysis looking for how many nulls in dataframe**

Find how many missed timing matts

In [124]:
missed5k = df[(df['5k min/mile1']==0.00)]

In [125]:
len(missed5k)

84

84 missing first 5k split

In [126]:
missed10k = df[(df['10k min/mile1']==0.00)]
len(missed10k)

84

In [127]:
missed15k = df[(df['15k min/mile1']==0.00)]
len(missed15k)

8

In [128]:
missed20k = df[(df['20k min/mile1']==0.00)]
len(missed20k)

5

In [129]:
missedhalf = df[(df['Half min/mile1']==0.00)]
len(missedhalf)

5

In [130]:
missed25k = df[(df['25k min/mile1']==0.00)]
len(missed25k)

3

In [131]:
missed30k = df[(df['30k min/mile1']==0.00)]
len(missed30k)

3

In [132]:
missed35k = df[(df['35k min/mile1']==0.00)]
len(missed35k)

3

In [133]:
missed40k = df[(df['40k min/mile1']==0.00)]
len(missed40k)

2

Only 1 person has all splits missing

In [135]:
df.head()

Name  Place Overall sex  BIB  5k min/mile1  \
0  Cherono, Lawrence (KEN)              1   M    4          4.45   
1     Debela, Dejene (ETH)              2   M   38          4.45   
2     Mengstu, Asefa (ETH)              3   M    5          4.45   
3      Karoki, Bedan (KEN)              4   M    9          4.45   
4       Abdi, Bashir (BEL)              5   M   10          4.46   

   10k min/mile1  15k min/mile1  20k min/mile1  Half min/mile1  25k min/mile1  \
0           4.45           4.45           4.47            4.45           4.50   
1           4.45           4.45           4.47            4.44           4.49   
2           4.45           4.45           4.47            4.44           4.49   
3           4.44           4.45           4.47            4.44           4.49   
4           4.44           4.48           4.53            4.57           4.57   

   30k min/mile1  35k min/mile1  40k min/mile1  Finish min/mile1  \
0           4.51           4.48           4.55              4.50   
1           4.51           4.48           4.55              4.52   
2           4.51           4.48           4.55              4.53   
3           4.52           4.48           4.55              4.57   
4           5.00           4.48           4.43              4.40   

   Total Finish time Start Time Finish Time of Day Finish Time.1  
0           2.095833   07:30:03         09:35:47AM      02:05:45  
1           2.096111   07:30:03         09:35:48AM      02:05:46  
2           2.096667   07:30:03         09:35:50AM      02:05:48  
3           2.098056   07:30:03         09:35:55AM      02:05:53  
4           2.103889   07:30:03         09:36:16AM      02:06:14

In [67]:
df2 = df2.sort_values('Place Overall')

In [68]:
df2.head(50)

Unnamed: 0                      Name    BIB Age Group Age  \
0               0   Cherono, Lawrence (KEN)      4     30-34  31   
1               1      Debela, Dejene (ETH)     38     20-24  24   
2               2      Mengstu, Asefa (ETH)      5     30-34  31   
3               3       Karoki, Bedan (KEN)      9     25-29  29   
4               4        Abdi, Bashir (BEL)     10     30-34  30   
5               5         Tura, Seifu (ETH)     39     20-24  22   
6               6     Chumba, Dickson (KEN)      6     30-34  32   
7               7           Farah, Mo (GBR)      1     35-39  36   
8               8        Riley, Jacob (USA)     16     30-34  30   
9               9       Mock, Jerrell (USA)     36     20-24  24   
10             10     Stinson, Parker (USA)     23     25-29  27   
11             11  Bumbalough, Andrew (USA)     20     30-34  32   
12             12       Oishi, Minato (JPN)     11     30-34  31   
13             13   Mcdonald, Matthew (USA)     21     25-29  26   
14             14        Smith, Scott (USA)     15     20-24  23   
15             15      Gregg, Brendan (USA)     17     30-34  30   
16             16        Droddy, Noah (USA)     25     25-29  29   
17             17    Given, Wilkerson (USA)     24     25-29  28   
18             18      Estrada, Diego (USA)     19     25-29  29   
19             19   Hayakawa, Tsubasa (JPN)     14     25-29  29   
20             20     Takeuchi, Ryoma (JPN)     12     25-29  27   
21             21      Shrader, Brian (USA)     18     25-29  28   
24102           0      Kosgei, Brigid (KEN)    101     25-29  25   
22             22        Monroe, Alex (USA)     34     25-29  27   
23             23     Kremske, Daniel (USA)     31     30-34  30   
24             24       Medina, Oscar (MEX)    568     25-29  27   
25             25   Campbell, Patrick (USA)    407     20-24  24   
26             26   Lustgarten, Brett (USA)     32     25-29  29   
27             27         Smith, Ryan (USA)    651     20-24  24   
28             28          Gade, Noah (USA)    465     20-24  24   
29             29          Bang, Eric (CAN)    370     25-29  29   
30             30      Eaton, Michael (USA)     29     30-34  32   
31             31       Thomas, Jason (USA)    665     25-29  25   
32             32       Nally, Daniel (USA)    580     30-34  34   
24103           1    Yeshaneh, Ababel (ETH)    106     25-29  28   
24104           2       Burka, Gelete (ETH)    102     30-34  33   
33             33   Craighead, Daniel (USA)    430     25-29  29   
34             34    Hillard, Matthew (USA)    497     30-34  30   
35             35       Chorney, Matt (USA)    417     30-34  31   
36             36   Bramblett, Jordan (USA)  13197     20-24  22   
37             37     Erskine, Andrew (USA)    457     25-29  27   
38             38       Nelson, Aaron (USA)    582     25-29  27   
39             39     Burkhardt, Glen (USA)    402     25-29  26   
40             40     Guillen, Ramiro (USA)    475     35-39  37   
41             41     Ripley, Zachary (USA)    619     30-34  31   
42             42     Barker, Zachary (USA)    371     25-29  27   
43             43      Hall, Benjamin (USA)    481     30-34  30   
44             44  Rylaarsdam, Andrew (USA)    713     20-24  24   
45             45       Rayner, Kenny (USA)    613     25-29  29   
46             46        Ruth, Marius (NOR)   5060     25-29  26   

       Place gender Place Age  Place Overall Start Time Finish Time  ...  \
0                 1         1              1   07:30:03    02:05:45  ...   
1                 2         1              2   07:30:03    02:05:46  ...   
2                 3         2              3   07:30:03    02:05:48  ...   
3                 4         1              4   07:30:03    02:05:53  ...   
4                 5         3              5   07:30:03    02:06:14  ...   
5                 6         2              6   07:30:03    02:08:35  ...   
6     

In [72]:
df2.to_csv('full_uneconded_results.csv')

In [70]:
missing = df2[df2.values  == '–']
len(missing)

795

In [73]:
df2.to_csv('Chi_final.csv')

In [74]:
df2.shape

(44672, 63)

In [75]:
df3.to_csv('final.csv')

In [80]:
df3 = df3.sort_values('Place Overall')

In [83]:
df3.head()

Unnamed: 0                     Name  BIB Age Group Age  Place gender  \
0           0  Cherono, Lawrence (KEN)    4     30-34  31             1   
1           1     Debela, Dejene (ETH)   38     20-24  24             2   
2           2     Mengstu, Asefa (ETH)    5     30-34  31             3   
3           3      Karoki, Bedan (KEN)    9     25-29  29             4   
4           4       Abdi, Bashir (BEL)   10     30-34  30             5   

  Place Age  Place Overall Start Time Finish Time  ... 5k min/mile1  \
0         1              1   07:30:03    02:05:45  ...        04.45   
1         1              2   07:30:03    02:05:46  ...        04.45   
2         2              3   07:30:03    02:05:48  ...        04.45   
3         1              4   07:30:03    02:05:53  ...        04.45   
4         3              5   07:30:03    02:06:14  ...        04.46   

  10k min/mile1 15k min/mile1 20k min/mile1 Half min/mile1 25k min/mile1  \
0         04.45         04.45         04.47          04.45           4.5   
1         04.45         04.45         04.47          04.44          4.49   
2         04.45         04.45         04.47          04.44          4.49   
3         04.44         04.45         04.47          04.44          4.49   
4         04.44         04.48         04.53          04.57          4.57   

  30k min/mile1 35k min/mile1 40k min/mile1 Finish min/mile1  
0          4.51          4.48          4.55              4.5  
1          4.51          4.48          4.55             4.52  
2          4.51          4.48          4.55             4.53  
3          4.52          4.48          4.55             4.57  
4             5          4.48          4.43              4.4  

[5 rows x 73 columns]

In [82]:
df3.to_csv('final.csv')

In [93]:
missing = df3[df3.values  == '–']
len(missing)

0

In [94]:
missing = df3[df3.values  == "00:00"]
len(missing)

0

In [97]:
df3.isnull().sum()

Unnamed: 0          0
Name                0
BIB                 0
Age Group           0
Age                 0
                   ..
25k min/mile1       0
30k min/mile1       0
35k min/mile1       0
40k min/mile1       0
Finish min/mile1    0
Length: 73, dtype: int64

In [5]:
df = pd.read_csv('Chi_final.csv')

/Users/johnvictorifermo/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (5,7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [6]:
df.head()

Unnamed: 0  Unnamed: 0.1                     Name  BIB Age Group Age  \
0           0             0  Cherono, Lawrence (KEN)    4     30-34  31   
1           1             1     Debela, Dejene (ETH)   38     20-24  24   
2           2             2     Mengstu, Asefa (ETH)    5     30-34  31   
3           3             3      Karoki, Bedan (KEN)    9     25-29  29   
4           4             4       Abdi, Bashir (BEL)   10     30-34  30   

   Place gender Place Age  Place Overall Start Time  ... 40k Time of Day  \
0             1         1              1   07:30:03  ...      09:29:12AM   
1             2         1              2   07:30:03  ...      09:29:10AM   
2             3         2              3   07:30:03  ...      09:29:10AM   
3             4         1              4   07:30:03  ...      09:29:11AM   
4             5         3              5   07:30:03  ...      09:29:55AM   

   40k Time 40k Difference 40k min/mile  Finish Finish Time of Day  \
0  01:59:10          15:17        04:55  Finish         09:35:47AM   
1  01:59:08          15:15        04:55  Finish         09:35:48AM   
2  01:59:08          15:15        04:55  Finish         09:35:50AM   
3  01:59:09          15:16        04:55  Finish         09:35:55AM   
4  01:59:53          14:39        04:43  Finish         09:36:16AM   

  Finish Time.1 Finish Difference Finish min/mile sex  
0      02:05:45             06:36           04:50   M  
1      02:05:46             06:39           04:52   M  
2      02:05:48             06:40           04:53   M  
3      02:05:53             06:45           04:57   M  
4      02:06:14             06:21           04:40   M  

[5 rows x 64 columns]

In [7]:
df.shape

(44672, 64)

Chi_final.csv will be final csv used for Tableau Analysis